keras_cv/models/resnet_v2.py

```diff
@@ -390,15 +390,45 @@ def as_backbone(self):
         The output of the backbone model will be a dict with int as key and tensor as
         value. The int key represent the level of the feature output.
 
+        Args:
+            min_level: optional int, the lowest level of feature to be included in the
+                output. Default to model's lowest feature level (based on the model structure).
+            max_level: optional int, the highest level of feature to be included in the
+                output. Default to model's highest feature level (based on the model structure).
+
         Returns:
             a `tf.keras.Model` which has dict as outputs.
         Raises:
             ValueError: When the model is lack of information for feature level, and can't
-            be converted to backbone model
+            be converted to backbone model, or the min_level/max_level param is out of
+            range based on the model structure.
         """
         if hasattr(self, "_backbone_level_outputs"):
             backbone_level_outputs = self._backbone_level_outputs
-            return tf.keras.Model(inputs=self.inputs, outputs=backbone_level_outputs)
+            model_levels = list(sorted(backbone_level_outputs.keys()))
```

Comment:
I think the `as_backbone` function here can be extract into a util function, and it will require model to populate certain attributes to work. Let's keep it in the Resnet for now (I would like to see if this can be generically apply to other models, eg non-functional models).

Will add a TODO item here to extract this.